src/waypoint.jsx

```diff
@@ -333,13 +352,20 @@ export default class Waypoint extends React.Component {
    * @return {Object}
    */
   render() {
-    // We need an element that we can locate in the DOM to determine where it is
-    // rendered relative to the top of its context.
-    return <span ref={this.refElement} style={{ fontSize: 0 }} />;
+    if (this.props.children) {
+      // TODO: consider returning children directly when there is only one child,
+      // using React.cloneElement to attach a ref (but child might already have a ref)
```

Comment:
You're right, @rattrayalex  – it's not currently possible. I'm thinking about the future when [Fiber](https://github.com/acdlite/react-fiber-architecture) supports it. The current code prepares us better for that day.